[MIG] sale_commission: Migrated to 10.0

hurrinico opened the pull request:
Porting of module sale_commission to Version 10

Commit:
[MIG] sale_commission: : Migrated to 10.0

FIX after changing customer on SO with lines, can't save sale order anymore
See https://github.com/odoo/odoo/issues/17618
Also add button to recompute agents on SO lines

hurrinico commented:
Still WIP

tafaRU commented on sale_commission/models/sale_commission.py:
```diff
@@ -47,9 +47,10 @@ class SaleCommissionSection(models.Model):
-        if self.amount_to < self.amount_from:
-            raise exceptions.ValidationError(
-                _("The lower limit cannot be greater than upper one."))
+        for section in self:
```
commission I think it's more suitable than section, what do you think?

hurrinico replied:
this model isn't sale commission but sale.commission.section, so maybe section is correct.

tafaRU replied:
ok, that's right!

tafaRU commented on sale_commission/models/sale_commission.py:
```diff
@@ -47,9 +47,10 @@ class SaleCommissionSection(models.Model):
-            raise exceptions.ValidationError(
-                _("The lower limit cannot be greater than upper one."))
+        for section in self:
+            if self.amount_to < self.amount_from:
```
s/self/section(or commission)

tafaRU reviewed (changes requested):
In addition please note the `<data>` tag in xml files can be removed.

Thanks!

pedrobaeza commented:
Hi, I'm going to work these days on finishing 9.0 migration with the latest changes including the web_widget_one2many_tags improvement, so maybe it's better to wait a bit for this migration and review the others. What do you think?

hurrinico commented:
@pedrobaeza i agree. When you finish the web_widget_one2many_tags porting we can improve this one and merge it

eLBati commented:
As workaround for https://github.com/odoo/odoo/issues/17618 I propose https://github.com/hurrinico/commission/pull/2

rgarnau reviewed (approved):
We have made an improvement for this PR in order to prepare agents' data outside from the method recompute_lines_agents.
See https://github.com/hurrinico/commission/pull/4.

@jbeficent

rgarnau committed:
add method to prepare agents

add same method for account invoice

eLBati reviewed (changes requested):
@hurrinico please fix
```
sale_commission/models/account_invoice.py:55:1: W293 blank line contains whitespace
```
and squash merge commits

hurrinico commented:
@eLBati travis is green

pedrobaeza commented:
Hi, guys, sorry for not answering about the update on 9.0, but I haven't been able to do it. This is still valid if there's no bug embedding a one2many inside a one2many (on v8 it was).

hurrinico commented:
@pedrobaeza there is a bug opened by @eLBati  odoo/odoo#17618, that is fixed by https://github.com/hurrinico/commission/pull/2

eLBati commented:
Hi @rgarnau , did you configure your github account?
I see in commit
https://github.com/OCA/commission/pull/117/commits/0ddb38650042800a67216483a9179703fbdec45c
you are not recognized.
Thanks

rgarnau commented:
Hi @eLBati - yes, already fixed. However, since this commit is already merged, I cannot amend the author anymore...

pedrobaeza commented:
Thanks for the info. It's still a bit uglier than with web_widget_one2many_tags, but it if works, perfect to continue this way.

hurrinico commented:
@rgarnau fixed.
i had conflicts try to fix flake 8 on your pr, so i made that mistake

eLBati commented:
@rgarnau about your commits in this PR, github is saying
> Unrecognized author. If this is you, make sure rgarnau@gmail.com is associated with your account. Click to add email addresses in your account settings

rgarnau commented:
@eLBati As I don't have permissions in this repo, I cannot amend the author of my commits. It's already fixed in my pull request.

eLBati commented:
@rgarnau is rgarnau@gmail.com  associated with your account?

rgarnau commented:
@eLBati no, that was a mistake. Instead, my account is rgarnau20@gmail.com

eLBati commented:
Ok, so @hurrinico please edit commits form @rgarnau fixing the email address. Thanks

eLBati commented:
@hurrinico @rgarnau done

eLBati committed:
FIX sale_commission quaterly settlement

as May would have June as period start:
((5 - 1) // 3 + 1) * 3

instead, the new formula:
>>> for m in range(1, 13):
...     print (m - 1) // 3 * 3 + 1
...
1
1
1
4
4
4
7
7
7
10
10
10

See
https://github.com/hurrinico/commission/pull/2
and
https://github.com/odoo/odoo/issues/17618

This extends the workaround, fixing "Record not found" after changing invoice date, for example

eLBati commented:
Ok, fixed quaterly settlement with https://github.com/OCA/commission/pull/117/commits/9279e38a7038a06f423b23f6ba5a66f98af660bc

atchuthan reviewed (approved):
Functional Test in Runbot :+1:

hurrinico commented:
@pedrobaeza @eLBati @atchuthan i can't remember what remains to do here

pedrobaeza commented:
@hurrinico nothing more as I can see. Just squash your last 3 commits together and then I'll merge.

eLBati commented:
@pedrobaeza I squashed the last 2 commits only, to preserve authorship

eLBati commented:
@pedrobaeza please don't merge because we found other issues with ```agents``` o2m

hendrixcosta committed:
[ADD] Tradução pt_BR

Commit:
Merge pull request #6 from kmee/10.0-mig-sale_commission

[ADD] translate for pt_BR

hurrinico commented:
IMHO after @eLBati commit this pr can be merged

atchuthan commented:
I agree with @hurrinico suggestion to merge this PR.
We could add the fix for o2m in a new PR

JordiBForgeFlow reviewed (changes requested):
We have issues when creating a demo database when the demo data tries to load a record expecting languange en_US to exist. See:
https://github.com/OCA/commission/pull/117/files#diff-855c05c79944f0f9cab407e8e2451171R51

eLBati commented:
@jbeficent do you suggest to remove line(s)
```
<field name="lang">en_US</field>
```
?

atchuthan commented:
@eLBati 
you could remove the line

Case scenario to reproduce:
- create demo database with another language except for en_US
- install this module and it will provide error

hurrinico commented:
@atchuthan i've removed the lines

atchuthan reviewed (approved):
:+1:

@jbeficent: please check and review your test case

pedrobaeza commented:
@jbeficent's case is covered already, so I merge this one finally.
